src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -87,8 +87,13 @@ void shape_infer(const Squeeze* op,
             std::copy_if(arg_shape.cbegin(), arg_shape.cend(), back_inserter(out_dims), not_squeezable_at_axis);
         }
         // When arg shape has got static rank but shape is dynamic and output shape dimensions is empty
-        // make dynamic output.
-        output_shape = arg_shape.is_dynamic() && out_dims.empty() ? PartialShape::dynamic() : T(out_dims);
+        // make dynamic output except the case of the rank of arg shape is 1 and 0 <= arg_shape[0] <= 1.
+        if (arg_shape.is_dynamic() && out_dims.empty()) {
```

Comment:
Add test scenario in `type_prop/squeeze.cpp` to cover this changes. It should fail without this change.
Try use test inputs like `PartialShape{0}, PartialShape{1}, PartialShape{{0,1}}`

Consider update comment here that this because of cpu_plugin which not handle dynamic rank shape.

Reply:
updated